better end call docs (#784)
elevenlabs/elevenlabs-docs: fern/conversational-ai/pages/customization/tools/end-call.mdx

subtitle: Let your agent automatically hang up on the user.
---

<Warning>

>   The **End Call** tool is added to agents created in the UI by default. For agents created via

Reviewer: ```suggestion
  The **End Call** tool is added to agents created in the ElevenLabs dashboard by default. For agents created via
```

>   API/SDK, you must add it manually as a system tool in your agent configuration. [See API

Reviewer: ```suggestion
  API or SDK, you must add it manually as a system tool in your agent configuration. [See API
```

> you must

Is it required?

Author: fixed in [99d68f3](https://github.com/elevenlabs/elevenlabs-docs/pull/784/commits/99d68f32a6e0e5399ddf0ecd44fb6ad209fd9052)

subtitle: Let your agent automatically hang up on the user.
---

<Warning>
  The **End Call** tool is added to agents created in the ElevenLabs dashboard by default. For

>   agents created via API or SDK, if you would like to enable the end_call tool, you must add it

Reviewer: ```suggestion
  agents created via API or SDK, if you would like to enable the End Call tool, you must add it
```

  - End the call if the user says "goodbye."
  - Conclude the call when a specific task is completed.

### API Implementation

> When creating an agent via API, add the End Call tool to your agent configuration. It must be defined as a system tool:

Reviewer: End Call is optional right? This makes it seem like it's a required tool, which if it is why don't we make it included by default?

Author: fixed

<CodeBlocks>

```python
from elevenlabs import (
    ConversationalConfig,
    ElevenLabs,
    AgentConfig,
    PromptAgent,
    PromptAgentToolsItem_System
)

# Initialize the client
client = ElevenLabs(api_key="YOUR_API_KEY")

# Create the end call tool
end_call_tool = PromptAgentToolsItem_System(
    name="end_call",
    description=""  # Optional: Customize when the tool should be triggered
)

# Create the agent configuration
conversation_config = ConversationalConfig(
    agent=AgentConfig(
        prompt=PromptAgent(
            tools=[end_call_tool]
        )
    )
)

# Create the agent
response = client.conversational_ai.create_agent(
    conversation_config=conversation_config
)
```

```javascript
import { ElevenLabs } from 'elevenlabs';

// Initialize the client
const client = new ElevenLabs({
  apiKey: 'YOUR_API_KEY',
});

// Create the agent with end call tool
await client.conversationalAi.createAgent({
  conversation_config: {
    agent: {
      prompt: {
        tools: [
          {
            type: 'system',

>             name: 'end_call',

Reviewer: I wish this were a list of enums rather than a string.

Author: im torn, definitely needs to strike balance between being able to add a tool for a single agent easily, but also reuse existing tools given that we're flattening tools to be sharable across agents

Author: will resolve for now so can merge this change but thinking about this
